server/routes/shared/action-plan/actionPlanPresenter.ts

```diff
@@ -68,19 +70,39 @@ export default class ActionPlanPresenter {
     return this.userType === 'probation-practitioner' && !this.actionPlanSummaryPresenter.actionPlanSubmitted
   }
 
-  get actionPlanVersions(): { approvalDate: string; versionNumber: number }[] {
+  get actionPlanVersions(): { approvalDate: string; versionNumber: number; href: string | null }[] {
     return this.approvedActionPlanSummaries
       .sort((summaryA, summaryB) => new Date(summaryB.approvedAt).getTime() - new Date(summaryA.approvedAt).getTime())
       .map((summary, index) => ({
         versionNumber: this.approvedActionPlanSummaries.length - index,
         approvalDate: dateUtils.formattedDate(summary.approvedAt, { month: 'short' }),
+        href: this.determineActionPlanHref(summary.id),
       }))
   }
 
+  private determineActionPlanHref(summaryId: string): string | null {
```

Comment:
I personally would prefer that PP and SP logic is less confusing by having the switch logic directly on the element (so SP would return null or empty).
Though I realise that it's not directly part of your PR since the `showActionPlanVersions` already exists.

Reply:
I've spent a bit of time trying to figure out an alternative, but have ended up keeping this the same - I've moved up the function `showActionPlanVersions` to keep it closer to the main function so it's a bit clearer when this will be called, but I think a newcomer to the codebase will probably check the `.njk` template first to see what this logic is, then work backwards to the presenter, rather than the other way round.

I feel like this PP-only restriction might not be a permanent thing, and it might be visible to SPs soon, so I'm going to leave it for now, unless you have any strong objections. We can always refactor it in future.